{"cells":[{"cell_type":"markdown","metadata":{"id":"O8Xz-lzkc1Ii"},"source":["### Cargar drive"]},{"cell_type":"code","execution_count":49,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":1147,"status":"ok","timestamp":1701140693439,"user":{"displayName":"Alberto Horacio Orozco Ramos","userId":"05995070409390431235"},"user_tz":360},"id":"xdYtBfEycmk3","outputId":"6cea283d-cdf5-4088-b037-8e4518e4b0c8"},"outputs":[{"output_type":"stream","name":"stdout","text":["Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount(\"/content/drive\", force_remount=True).\n"]}],"source":["from google.colab import drive\n","drive.mount('/content/drive')"]},{"cell_type":"markdown","metadata":{"id":"2XiABHsUdroC"},"source":["### Installing dependencies"]},{"cell_type":"code","execution_count":50,"metadata":{"id":"m5yqvGSKdrwP","executionInfo":{"status":"ok","timestamp":1701140699037,"user_tz":360,"elapsed":5603,"user":{"displayName":"Alberto Horacio Orozco Ramos","userId":"05995070409390431235"}},"colab":{"base_uri":"https://localhost:8080/"},"outputId":"58cff4fa-2bcd-431b-bf71-e9ca7b7e8264"},"outputs":[{"output_type":"stream","name":"stdout","text":["Requirement already satisfied: mediapipe in /usr/local/lib/python3.10/dist-packages (0.10.8)\n","Requirement already satisfied: face_recognition in /usr/local/lib/python3.10/dist-packages (1.3.0)\n","Requirement already satisfied: opencv-python in /usr/local/lib/python3.10/dist-packages (4.7.0.72)\n","Requirement already satisfied: dlib==19.24.2 in /usr/local/lib/python3.10/dist-packages (19.24.2)\n","Requirement already satisfied: yolov5 in /usr/local/lib/python3.10/dist-packages (7.0.13)\n","Requirement already satisfied: absl-py in /usr/local/lib/python3.10/dist-packages (from mediapipe) (1.4.0)\n","Requirement already satisfied: attrs>=19.1.0 in /usr/local/lib/python3.10/dist-packages (from mediapipe) (23.1.0)\n","Requirement already satisfied: flatbuffers>=2.0 in /usr/local/lib/python3.10/dist-packages (from mediapipe) (23.5.26)\n","Requirement already satisfied: matplotlib in /usr/local/lib/python3.10/dist-packages (from mediapipe) (3.7.1)\n","Requirement already satisfied: numpy in /usr/local/lib/python3.10/dist-packages (from mediapipe) (1.23.5)\n","Requirement already satisfied: opencv-contrib-python in /usr/local/lib/python3.10/dist-packages (from mediapipe) (4.8.0.76)\n","Requirement already satisfied: protobuf<4,>=3.11 in /usr/local/lib/python3.10/dist-packages (from mediapipe) (3.20.3)\n","Requirement already satisfied: sounddevice>=0.4.4 in /usr/local/lib/python3.10/dist-packages (from mediapipe) (0.4.6)\n","Requirement already satisfied: face-recognition-models>=0.3.0 in /usr/local/lib/python3.10/dist-packages (from face_recognition) (0.3.0)\n","Requirement already satisfied: Click>=6.0 in /usr/local/lib/python3.10/dist-packages (from face_recognition) (8.1.7)\n","Requirement already satisfied: Pillow in /usr/local/lib/python3.10/dist-packages (from face_recognition) (9.4.0)\n","Requirement already satisfied: gitpython>=3.1.30 in /usr/local/lib/python3.10/dist-packages (from yolov5) (3.1.40)\n","Requirement already satisfied: psutil in /usr/local/lib/python3.10/dist-packages (from yolov5) (5.9.5)\n","Requirement already satisfied: PyYAML>=5.3.1 in /usr/local/lib/python3.10/dist-packages (from yolov5) (6.0.1)\n","Requirement already satisfied: requests>=2.23.0 in /usr/local/lib/python3.10/dist-packages (from yolov5) (2.31.0)\n","Requirement already satisfied: scipy>=1.4.1 in /usr/local/lib/python3.10/dist-packages (from yolov5) (1.11.3)\n","Requirement already satisfied: thop>=0.1.1 in /usr/local/lib/python3.10/dist-packages (from yolov5) (0.1.1.post2209072238)\n","Requirement already satisfied: torch>=1.7.0 in /usr/local/lib/python3.10/dist-packages (from yolov5) (2.1.0+cu118)\n","Requirement already satisfied: torchvision>=0.8.1 in /usr/local/lib/python3.10/dist-packages (from yolov5) (0.16.0+cu118)\n","Requirement already satisfied: tqdm>=4.64.0 in /usr/local/lib/python3.10/dist-packages (from yolov5) (4.66.1)\n","Requirement already satisfied: ultralytics>=8.0.100 in /usr/local/lib/python3.10/dist-packages (from yolov5) (8.0.219)\n","Requirement already satisfied: tensorboard>=2.4.1 in /usr/local/lib/python3.10/dist-packages (from yolov5) (2.14.1)\n","Requirement already satisfied: pandas>=1.1.4 in /usr/local/lib/python3.10/dist-packages (from yolov5) (1.5.3)\n","Requirement already satisfied: seaborn>=0.11.0 in /usr/local/lib/python3.10/dist-packages (from yolov5) (0.12.2)\n","Requirement already satisfied: setuptools>=65.5.1 in /usr/local/lib/python3.10/dist-packages (from yolov5) (67.7.2)\n","Requirement already satisfied: fire in /usr/local/lib/python3.10/dist-packages (from yolov5) (0.5.0)\n","Requirement already satisfied: boto3>=1.19.1 in /usr/local/lib/python3.10/dist-packages (from yolov5) (1.33.0)\n","Requirement already satisfied: sahi>=0.11.10 in /usr/local/lib/python3.10/dist-packages (from yolov5) (0.11.15)\n","Requirement already satisfied: huggingface-hub>=0.12.0 in /usr/local/lib/python3.10/dist-packages (from yolov5) (0.19.4)\n","Requirement already satisfied: roboflow>=0.2.29 in /usr/local/lib/python3.10/dist-packages (from yolov5) (1.1.9)\n","Requirement already satisfied: botocore<1.34.0,>=1.33.0 in /usr/local/lib/python3.10/dist-packages (from boto3>=1.19.1->yolov5) (1.33.0)\n","Requirement already satisfied: jmespath<2.0.0,>=0.7.1 in /usr/local/lib/python3.10/dist-packages (from boto3>=1.19.1->yolov5) (1.0.1)\n","Requirement already satisfied: s3transfer<0.9.0,>=0.8.0 in /usr/local/lib/python3.10/dist-packages (from boto3>=1.19.1->yolov5) (0.8.0)\n","Requirement already satisfied: gitdb<5,>=4.0.1 in /usr/local/lib/python3.10/dist-packages (from gitpython>=3.1.30->yolov5) (4.0.11)\n","Requirement already satisfied: filelock in /usr/local/lib/python3.10/dist-packages (from huggingface-hub>=0.12.0->yolov5) (3.13.1)\n","Requirement already satisfied: fsspec>=2023.5.0 in /usr/local/lib/python3.10/dist-packages (from huggingface-hub>=0.12.0->yolov5) (2023.6.0)\n","Requirement already satisfied: typing-extensions>=3.7.4.3 in /usr/local/lib/python3.10/dist-packages (from huggingface-hub>=0.12.0->yolov5) (4.5.0)\n","Requirement already satisfied: packaging>=20.9 in /usr/local/lib/python3.10/dist-packages (from huggingface-hub>=0.12.0->yolov5) (23.2)\n","Requirement already satisfied: contourpy>=1.0.1 in /usr/local/lib/python3.10/dist-packages (from matplotlib->mediapipe) (1.2.0)\n","Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.10/dist-packages (from matplotlib->mediapipe) (0.10.0)\n","Requirement already satisfied: fonttools>=4.22.0 in /usr/local/lib/python3.10/dist-packages (from matplotlib->mediapipe) (4.44.3)\n","Requirement already satisfied: kiwisolver>=1.0.1 in /usr/local/lib/python3.10/dist-packages (from matplotlib->mediapipe) (1.4.5)\n","Requirement already satisfied: pyparsing>=2.3.1 in /usr/local/lib/python3.10/dist-packages (from matplotlib->mediapipe) (2.4.7)\n","Requirement already satisfied: python-dateutil>=2.7 in /usr/local/lib/python3.10/dist-packages (from matplotlib->mediapipe) (2.8.2)\n","Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.10/dist-packages (from pandas>=1.1.4->yolov5) (2023.3.post1)\n","Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.10/dist-packages (from requests>=2.23.0->yolov5) (3.3.2)\n","Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/dist-packages (from requests>=2.23.0->yolov5) (2.10)\n","Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests>=2.23.0->yolov5) (2.0.7)\n","Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from requests>=2.23.0->yolov5) (2023.7.22)\n","Requirement already satisfied: chardet==4.0.0 in /usr/local/lib/python3.10/dist-packages (from roboflow>=0.2.29->yolov5) (4.0.0)\n","Requirement already satisfied: opencv-python-headless==4.8.0.74 in /usr/local/lib/python3.10/dist-packages (from roboflow>=0.2.29->yolov5) (4.8.0.74)\n","Requirement already satisfied: python-dotenv in /usr/local/lib/python3.10/dist-packages (from roboflow>=0.2.29->yolov5) (1.0.0)\n","Requirement already satisfied: six in /usr/local/lib/python3.10/dist-packages (from roboflow>=0.2.29->yolov5) (1.16.0)\n","Requirement already satisfied: supervision in /usr/local/lib/python3.10/dist-packages (from roboflow>=0.2.29->yolov5) (0.16.0)\n","Requirement already satisfied: requests-toolbelt in /usr/local/lib/python3.10/dist-packages (from roboflow>=0.2.29->yolov5) (1.0.0)\n","Requirement already satisfied: python-magic in /usr/local/lib/python3.10/dist-packages (from roboflow>=0.2.29->yolov5) (0.4.27)\n","Requirement already satisfied: shapely>=1.8.0 in /usr/local/lib/python3.10/dist-packages (from sahi>=0.11.10->yolov5) (2.0.2)\n","Requirement already satisfied: pybboxes==0.1.6 in /usr/local/lib/python3.10/dist-packages (from sahi>=0.11.10->yolov5) (0.1.6)\n","Requirement already satisfied: terminaltables in /usr/local/lib/python3.10/dist-packages (from sahi>=0.11.10->yolov5) (3.1.10)\n","Requirement already satisfied: CFFI>=1.0 in /usr/local/lib/python3.10/dist-packages (from sounddevice>=0.4.4->mediapipe) (1.16.0)\n","Requirement already satisfied: grpcio>=1.48.2 in /usr/local/lib/python3.10/dist-packages (from tensorboard>=2.4.1->yolov5) (1.59.2)\n","Requirement already satisfied: google-auth<3,>=1.6.3 in /usr/local/lib/python3.10/dist-packages (from tensorboard>=2.4.1->yolov5) (2.17.3)\n","Requirement already satisfied: google-auth-oauthlib<1.1,>=0.5 in /usr/local/lib/python3.10/dist-packages (from tensorboard>=2.4.1->yolov5) (1.0.0)\n","Requirement already satisfied: markdown>=2.6.8 in /usr/local/lib/python3.10/dist-packages (from tensorboard>=2.4.1->yolov5) (3.5.1)\n","Requirement already satisfied: tensorboard-data-server<0.8.0,>=0.7.0 in /usr/local/lib/python3.10/dist-packages (from tensorboard>=2.4.1->yolov5) (0.7.2)\n","Requirement already satisfied: werkzeug>=1.0.1 in /usr/local/lib/python3.10/dist-packages (from tensorboard>=2.4.1->yolov5) (3.0.1)\n","Requirement already satisfied: sympy in /usr/local/lib/python3.10/dist-packages (from torch>=1.7.0->yolov5) (1.12)\n","Requirement already satisfied: networkx in /usr/local/lib/python3.10/dist-packages (from torch>=1.7.0->yolov5) (3.2.1)\n","Requirement already satisfied: jinja2 in /usr/local/lib/python3.10/dist-packages (from torch>=1.7.0->yolov5) (3.1.2)\n","Requirement already satisfied: triton==2.1.0 in /usr/local/lib/python3.10/dist-packages (from torch>=1.7.0->yolov5) (2.1.0)\n","Requirement already satisfied: py-cpuinfo in /usr/local/lib/python3.10/dist-packages (from ultralytics>=8.0.100->yolov5) (9.0.0)\n","Requirement already satisfied: termcolor in /usr/local/lib/python3.10/dist-packages (from fire->yolov5) (2.3.0)\n","Requirement already satisfied: pycparser in /usr/local/lib/python3.10/dist-packages (from CFFI>=1.0->sounddevice>=0.4.4->mediapipe) (2.21)\n","Requirement already satisfied: smmap<6,>=3.0.1 in /usr/local/lib/python3.10/dist-packages (from gitdb<5,>=4.0.1->gitpython>=3.1.30->yolov5) (5.0.1)\n","Requirement already satisfied: cachetools<6.0,>=2.0.0 in /usr/local/lib/python3.10/dist-packages (from google-auth<3,>=1.6.3->tensorboard>=2.4.1->yolov5) (5.3.2)\n","Requirement already satisfied: pyasn1-modules>=0.2.1 in /usr/local/lib/python3.10/dist-packages (from google-auth<3,>=1.6.3->tensorboard>=2.4.1->yolov5) (0.3.0)\n","Requirement already satisfied: rsa<5,>=3.1.4 in /usr/local/lib/python3.10/dist-packages (from google-auth<3,>=1.6.3->tensorboard>=2.4.1->yolov5) (4.9)\n","Requirement already satisfied: requests-oauthlib>=0.7.0 in /usr/local/lib/python3.10/dist-packages (from google-auth-oauthlib<1.1,>=0.5->tensorboard>=2.4.1->yolov5) (1.3.1)\n","Requirement already satisfied: MarkupSafe>=2.1.1 in /usr/local/lib/python3.10/dist-packages (from werkzeug>=1.0.1->tensorboard>=2.4.1->yolov5) (2.1.3)\n","Requirement already satisfied: mpmath>=0.19 in /usr/local/lib/python3.10/dist-packages (from sympy->torch>=1.7.0->yolov5) (1.3.0)\n","Requirement already satisfied: pyasn1<0.6.0,>=0.4.6 in /usr/local/lib/python3.10/dist-packages (from pyasn1-modules>=0.2.1->google-auth<3,>=1.6.3->tensorboard>=2.4.1->yolov5) (0.5.0)\n","Requirement already satisfied: oauthlib>=3.0.0 in /usr/local/lib/python3.10/dist-packages (from requests-oauthlib>=0.7.0->google-auth-oauthlib<1.1,>=0.5->tensorboard>=2.4.1->yolov5) (3.2.2)\n"]}],"source":["!pip install mediapipe face_recognition opencv-python dlib==19.24.2 yolov5"]},{"cell_type":"markdown","metadata":{"id":"A0wgfnX_6TSa"},"source":["### Python Scripts Path"]},{"cell_type":"code","execution_count":51,"metadata":{"id":"3AMXq0i16Tdo","executionInfo":{"status":"ok","timestamp":1701140699038,"user_tz":360,"elapsed":27,"user":{"displayName":"Alberto Horacio Orozco Ramos","userId":"05995070409390431235"}},"colab":{"base_uri":"https://localhost:8080/"},"outputId":"5847b663-3e16-43eb-9045-d558877f9461"},"outputs":[{"output_type":"stream","name":"stdout","text":["/content/drive/MyDrive/Equipo 2 - TC3007C/Reto IA II/Modelo Participación y Asistencia\n"]}],"source":["%cd /content/drive/MyDrive/Equipo 2 - TC3007C/Reto IA II/Modelo Participación y Asistencia\n","# %cd /content/drive/MyDrive/Modelo Participación y Asistencia"]},{"cell_type":"markdown","metadata":{"id":"wtkhPfVrdarR"},"source":["### Library import"]},{"cell_type":"code","execution_count":52,"metadata":{"id":"JpmRRM6bda0t","executionInfo":{"status":"ok","timestamp":1701140699038,"user_tz":360,"elapsed":22,"user":{"displayName":"Alberto Horacio Orozco Ramos","userId":"05995070409390431235"}}},"outputs":[],"source":["# Libraries to use\n","import cv2\n","import yolov5\n","import face_recognition\n","import mediapipe as mp\n","from datetime import datetime\n","import traceback\n","from matplotlib import pyplot as plt\n","\n","# Auxiliar Scripts\n","from Person import Person"]},{"cell_type":"markdown","metadata":{"id":"WyiiBZmbC603"},"source":["### Define the students data"]},{"cell_type":"code","execution_count":53,"metadata":{"id":"ed0P-jf_C7Ay","executionInfo":{"status":"ok","timestamp":1701140699038,"user_tz":360,"elapsed":20,"user":{"displayName":"Alberto Horacio Orozco Ramos","userId":"05995070409390431235"}}},"outputs":[],"source":["data = {\n","    'person_1' : {\n","        'id': 1,\n","        'name': 'Alberto',\n","        # 'img_url': '/content/drive/MyDrive/Modelo Participación y Asistencia/20231113_110250.jpg',\n","        'img_url': '/content/drive/MyDrive/Equipo 2 - TC3007C/Reto IA II/Modelo Participación y Asistencia/Alberto.jpg',\n","    },\n","    'person_2' : {\n","        'id': 2,\n","        'name': 'Luis',\n","        # 'img_url': '/content/drive/MyDrive/Modelo Participación y Asistencia/20231001_184708.jpg',\n","        'img_url': '/content/drive/MyDrive/Equipo 2 - TC3007C/Reto IA II/Modelo Participación y Asistencia/Luis.jpg',\n","    },\n","    'person_3' : {\n","        'id': 3,\n","        'name' : 'Julian',\n","        # 'img_url': '/content/drive/MyDrive/Modelo Participación y Asistencia/Julian.jpg'\n","        'img_url': '/content/drive/MyDrive/Equipo 2 - TC3007C/Reto IA II/Modelo Participación y Asistencia/Julian.jpg'\n","    },\n","    'person_4' : {\n","        'id': 4,\n","        'name' : 'Antonio',\n","        # 'img_url': '/content/drive/MyDrive/Modelo Participación y Asistencia/Julian.jpg'\n","        'img_url': '/content/drive/MyDrive/Equipo 2 - TC3007C/Reto IA II/Modelo Participación y Asistencia/Antonio.jpg'\n","    }\n","\n","    # 'person_4' : {\n","    #     'id': 4,\n","    #     'name': 'Alonso',\n","    #     # 'img_url': '/content/drive/MyDrive/Modelo Participación y Asistencia/20230805_113046.jpg'\n","    #     'img_url': '/content/drive/MyDrive/Equipo 2 - TC3007C/Reto IA II/Modelo Participación y Asistencia/20230805_113046.jpg',\n","    # }\n","}"]},{"cell_type":"markdown","metadata":{"id":"IHHx1X6chKA1"},"source":["### Create arrays to store people, face names and face encodings"]},{"cell_type":"code","execution_count":54,"metadata":{"id":"I3glZWYjhNyT","executionInfo":{"status":"ok","timestamp":1701140699039,"user_tz":360,"elapsed":20,"user":{"displayName":"Alberto Horacio Orozco Ramos","userId":"05995070409390431235"}}},"outputs":[],"source":["people = []\n","known_face_encodings = []"]},{"cell_type":"markdown","metadata":{"id":"IlNdXsyphTJ-"},"source":["### Initialize an object Person to keep track of attendance and participation"]},{"cell_type":"code","execution_count":55,"metadata":{"id":"h6J0uAtShTSz","executionInfo":{"status":"ok","timestamp":1701140706220,"user_tz":360,"elapsed":7199,"user":{"displayName":"Alberto Horacio Orozco Ramos","userId":"05995070409390431235"}}},"outputs":[],"source":["for person_key, person_info in data.items():\n","    name = person_info['name']\n","    person_id = person_info['id']\n","    img_url = person_info['img_url']\n","\n","    person = Person(name, person_id, img_url)\n","    people.append(person)\n","    known_face_encodings.append(person.person_face_encoding)"]},{"cell_type":"markdown","metadata":{"id":"yg9fXsU68tIP"},"source":["### Initialize Pose Model"]},{"cell_type":"code","execution_count":56,"metadata":{"id":"3PtNYGRM8tNu","executionInfo":{"status":"ok","timestamp":1701140706221,"user_tz":360,"elapsed":34,"user":{"displayName":"Alberto Horacio Orozco Ramos","userId":"05995070409390431235"}}},"outputs":[],"source":["pose = mp.solutions.pose.Pose(min_detection_confidence=0.5, min_tracking_confidence=0.9)"]},{"cell_type":"markdown","metadata":{"id":"5cYhgtGO81HI"},"source":["### Load the YOLOv5 model"]},{"cell_type":"code","execution_count":57,"metadata":{"id":"gM9rXUpL81OZ","executionInfo":{"status":"ok","timestamp":1701140707145,"user_tz":360,"elapsed":956,"user":{"displayName":"Alberto Horacio Orozco Ramos","userId":"05995070409390431235"}}},"outputs":[],"source":["model_path = 'yolov5s.pt'\n","# device = \"cpu\"  # for cpu\n","device = 0  # for gpu\n","yolov5 = yolov5.YOLOv5(model_path, device, load_on_init=True)"]},{"cell_type":"markdown","metadata":{"id":"U_l5MNMEhjan"},"source":["### Path to the input MP4 file, Creation of the VideoCapture Object and Ouput file path"]},{"cell_type":"code","execution_count":58,"metadata":{"id":"dowBIxw8hjjW","executionInfo":{"status":"ok","timestamp":1701140707732,"user_tz":360,"elapsed":592,"user":{"displayName":"Alberto Horacio Orozco Ramos","userId":"05995070409390431235"}}},"outputs":[],"source":["# Input file path\n","mp4_input_file_path = '/content/drive/MyDrive/Equipo 2 - TC3007C/Reto IA II/Modelo Participación y Asistencia/input_5.mp4'\n","# mp4_input_file_path = '/content/drive/MyDrive/Modelo Participación y Asistencia/input.mp4'\n","# Create a VideoCapture object using the MP4 file\n","video_capture = cv2.VideoCapture(mp4_input_file_path)\n","# Output filepath\n","mp4_output_file_path = '/content/drive/MyDrive/Equipo 2 - TC3007C/Reto IA II/Modelo Participación y Asistencia/output_video_10.mp4'\n","# mp4_output_file_path = '/content/drive/MyDrive/Modelo Participación y Asistencia/output_video_7.mp4'"]},{"cell_type":"markdown","metadata":{"id":"Q8TPxD9gii_z"},"source":["### Define the codec and VideoWriter to generate an output MP4 file"]},{"cell_type":"code","execution_count":59,"metadata":{"id":"aEUvoOVWijIX","executionInfo":{"status":"ok","timestamp":1701140707990,"user_tz":360,"elapsed":265,"user":{"displayName":"Alberto Horacio Orozco Ramos","userId":"05995070409390431235"}}},"outputs":[],"source":["# Get the video frame width, height, and frames per second information\n","width = int(video_capture.get(cv2.CAP_PROP_FRAME_WIDTH))\n","height = int(video_capture.get(cv2.CAP_PROP_FRAME_HEIGHT))\n","fps = int(video_capture.get(cv2.CAP_PROP_FPS))\n","\n","# Define the codec and create a VideoWriter object\n","fourcc = cv2.VideoWriter_fourcc(*'mp4v')  # Use 'XVID' for AVI or 'mp4v' for MP4\n","out = cv2.VideoWriter(mp4_output_file_path, fourcc, fps, (width, height))"]},{"cell_type":"markdown","metadata":{"id":"iaJQ6LtUiafP"},"source":["### While loop for the facial and pose recognition"]},{"cell_type":"code","execution_count":60,"metadata":{"id":"vhi8CcwnialF","executionInfo":{"status":"ok","timestamp":1701141133345,"user_tz":360,"elapsed":425361,"user":{"displayName":"Alberto Horacio Orozco Ramos","userId":"05995070409390431235"}},"colab":{"base_uri":"https://localhost:8080/"},"outputId":"970f6cda-ecc1-48b2-a766-660dca3b618f"},"outputs":[{"output_type":"stream","name":"stdout","text":["Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Participation Registered for Alberto\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Alberto's arm down\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Participation Registered for Alberto\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Alberto's arm down\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Participation Registered for Antonio\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Participation Registered for Alberto\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Alberto's arm down\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Participation Registered for Alberto\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Error processing image: list index out of range\n","Traceback (most recent call last):\n","  File \"<ipython-input-60-b763a209dd3b>\", line 97, in <cell line: 2>\n","    rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","IndexError: list index out of range\n","\n","Frame is None. Exiting.\n"]}],"source":["# Process each frame of the video\n","while True:\n","    # Read the next frame\n","    success, frame = video_capture.read()\n","\n","    if frame is None:\n","      print(\"Frame is None. Exiting.\")\n","      break\n","\n","    # Find all face locations and face encodings in the current frame\n","    face_locations = face_recognition.face_locations(frame)\n","    face_encodings = face_recognition.face_encodings(frame, face_locations)\n","\n","    # Process each face in the frame\n","    for (top, right, bottom, left), face_encoding in zip(face_locations, face_encodings):\n","        # Check if the face matches any of the known persons\n","        matches = face_recognition.compare_faces(known_face_encodings, face_encoding, tolerance=0.5)\n","        name = \"Unknown\"\n","\n","        if True in matches:\n","            first_match_index = matches.index(True)\n","            person = people[first_match_index]\n","            name = person.name\n","\n","            if person.name != \"Unknown\" and not person.assistance:\n","                # Log attendance\n","                with open(\"attendance_log.txt\", \"a\") as log_file:\n","                    log_file.write(f\"{name} - {datetime.now()}\\n\")\n","\n","                # Mark attendance as taken\n","                person.assistance = True\n","\n","        # Draw rectangle around the face\n","        cv2.rectangle(frame, (left, top), (right, bottom), (0, 0, 255), 2)\n","\n","        # Draw label\n","        font = cv2.FONT_HERSHEY_DUPLEX\n","        cv2.putText(frame, name, (left + 6, bottom - 6), font, 0.5, (255, 255, 255), 1)\n","\n","    # Perform object detection on the frame\n","    results = yolov5.predict(frame, size=640, augment=False)\n","    detections = results.pred[0]\n","\n","    # Check whether the bounding box centroids are inside the ROI\n","    for detection in detections:\n","        xmin = detection[0]\n","        ymin = detection[1]\n","        xmax = detection[2]\n","        ymax = detection[3]\n","        score = detection[4]\n","        class_id = detection[5]\n","        centroid_x = int(xmin + xmax) // 2\n","        centroid_y = int(ymin + ymax) // 2\n","\n","        # Threshold score\n","        if score >= 0.6:\n","            if class_id == 0:\n","                color = (255, 0, 0)\n","                cv2.rectangle(frame, (int(xmin), int(ymin)), (int(xmax), int(ymax)), color, 1)\n","\n","                # Padding\n","                padding = 25\n","                person_roi = frame[int(ymin):int(ymax), int(xmin):int(xmax)]\n","\n","                try:  # Mediapipe\n","                    # Count arm raises\n","                    results_pose = pose.process(person_roi)\n","                    if results_pose.pose_landmarks:\n","\n","                        # Shoulder landmarks (left and right)\n","                        left_shoulder = results_pose.pose_landmarks.landmark[mp.solutions.pose.PoseLandmark.LEFT_SHOULDER]\n","                        right_shoulder = results_pose.pose_landmarks.landmark[mp.solutions.pose.PoseLandmark.RIGHT_SHOULDER]\n","\n","                        # Hand landmarks (for example, the tip of the index finger)\n","                        left_hand = results_pose.pose_landmarks.landmark[mp.solutions.pose.PoseLandmark.LEFT_INDEX]\n","                        right_hand = results_pose.pose_landmarks.landmark[mp.solutions.pose.PoseLandmark.RIGHT_INDEX]\n","\n","                        # Pose tracking logic (same as in your original script)\n","                        left_arm_landmarks = results_pose.pose_landmarks.landmark[mp.solutions.pose.PoseLandmark.LEFT_SHOULDER:mp.solutions.pose.PoseLandmark.LEFT_WRIST]\n","                        right_arm_landmarks = results_pose.pose_landmarks.landmark[mp.solutions.pose.PoseLandmark.RIGHT_SHOULDER:mp.solutions.pose.PoseLandmark.RIGHT_WRIST]\n","\n","                        left_arm_average_y = sum(point.y for point in left_arm_landmarks) / len(left_arm_landmarks)\n","                        right_arm_average_y = sum(point.y for point in right_arm_landmarks) / len(right_arm_landmarks)\n","\n","                        left_arm_up = left_arm_average_y < left_arm_landmarks[0].y\n","                        right_arm_up = right_arm_average_y < right_arm_landmarks[0].y\n","\n","                        # Check if the person is participating based on hand positions\n","                        not_participating = left_hand.y < left_shoulder.y and right_hand.y < right_shoulder.y\n","\n","                        rgb_roi = cv2.cvtColor(person_roi, cv2.COLOR_BGR2RGB)\n","\n","                        # cv2.imshow(\"Detected participation\", rgb_roi)\n","                        # cv2.waitKey(0)\n","\n","                        if rgb_roi.any():\n","                            rgb_encoding = face_recognition.face_encodings(rgb_roi)[0]\n","\n","                            # print(f\"left hand: {left_hand}, right_hand: {right_hand}\")\n","\n","                            matches = face_recognition.compare_faces(known_face_encodings, rgb_encoding, tolerance=0.48)\n","\n","                            if True in matches:\n","                                idx = matches.index(True)\n","\n","                                if people[idx].arm_raised == False and (left_arm_up or right_arm_up) and not not_participating:\n","                                        people[idx].participations += 1\n","                                        people[idx].arm_raised = True\n","                                        print(f\"Participation Registered for {people[idx].name}\")\n","\n","                                elif people[idx].arm_raised == True and not (left_arm_up or right_arm_up) and not_participating:\n","                                        print(f\"{people[idx].name}'s arm down\")\n","                                        people[idx].arm_raised = False\n","\n","                except Exception as e:\n","                    print(f\"Error processing image: {e}\")\n","                    print(traceback.format_exc())\n","\n","            else:\n","                pass\n","\n","    # Display the frame with counters for each person\n","    for idx, person in enumerate(people):\n","        cv2.putText(frame, f\"{person.name}: {person.participations} participations\", (10, 30 + 20 * idx),\n","            cv2.FONT_HERSHEY_DUPLEX, 0.5, (255, 0, 0), 1)\n","\n","    # Display the frame\n","    # cv2.imshow(\"Video\", frame)\n","\n","    out.write(frame)\n","    if cv2.waitKey(1) & 0xFF == ord('q'):\n","        break\n","\n","# Release the video capture object\n","video_capture.release()\n","out.release()\n","\n","cv2.destroyAllWindows()"]}],"metadata":{"accelerator":"GPU","colab":{"provenance":[],"gpuType":"T4"},"kernelspec":{"display_name":"Python 3","name":"python3"},"language_info":{"name":"python"}},"nbformat":4,"nbformat_minor":0}